src/pages/workspaces/workspace/analysis/modals/AnalysisModal.ts

```diff
@@ -18,45 +18,73 @@ import { FormLabel } from 'src/libs/forms'
 } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
-import { cloudAppTools, cloudRuntimeTools, getAppType, getToolFromFileExtension, getToolFromRuntime, isAppToolLabel, runtimeTools, toolExtensionDisplay, toolLabels, tools } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { AppTool, cloudAppTools, cloudRuntimeTools, ExtensionDisplay, getAppType, getToolFromFileExtension, getToolFromRuntime, isAppToolLabel, runtimeTools, Tool, toolExtensionDisplay, toolLabels, tools } from 'src/pages/workspaces/workspace/analysis/tool-utils'
 import validate from 'validate.js'
 
 
 const titleId = 'analysis-modal-title'
 const analysisMode = Symbol('artifact')
 const environmentMode = Symbol('environment')
 
+export interface AnalysisModalProps {
+  isOpen: boolean
+  workspace: BaseWorkspace
+  location: any
+  runtimes: any
+  apps: AppTool
+  appDataDisks: any
+  persistentDisks: any
+  onDismiss: () => void
+  onError: () => void
+  onSuccess: () => void
+  openUploader: () => void
+  uploadFiles: () => void
+  //TODO: Temporary until Analyses implements useAnalysisFiles
+  refreshAnalyses: () => void
+}
+
 export const AnalysisModal = withDisplayName('AnalysisModal')(
   ({
-    isOpen, onDismiss, onError, onSuccess, uploadFiles, openUploader, runtimes, apps, appDataDisks, refreshAnalyses,
-    analyses, workspace, persistentDisks, location, workspace: { workspace: { workspaceId, googleProject, bucketName } }
-  }) => {
-    const [viewMode, setViewMode] = useState(undefined)
-    const cloudProvider = getCloudProviderFromWorkspace(workspace)
+    isOpen, onDismiss, onError, onSuccess,
+    runtimes, apps,
+    appDataDisks,
+    persistentDisks,
+    uploadFiles,
+    openUploader,
+    workspace,
+    location,
+    refreshAnalyses
+  }: AnalysisModalProps) => {
+    const [viewMode, setViewMode] = useState<any>()
+    const cloudPlatform = workspace.workspace.cloudPlatform.toUpperCase()
     const [notebookKernel, setNotebookKernel] = useState('python3')
     const [analysisName, setAnalysisName] = useState('')
     const prevAnalysisName = usePrevious(analysisName)
-    const [currentToolObj, setCurrentToolObj] = useState(undefined)
+    const [currentToolObj, setCurrentToolObj] = useState<Tool>()
     const [fileExt, setFileExt] = useState('')
     const currentTool = currentToolObj?.label
 
-    const currentRuntime = getCurrentRuntime(runtimes)
+    const currentRuntime: any = getCurrentRuntime(runtimes)
     const currentDisk = getCurrentPersistentDisk(runtimes, persistentDisks)
     const currentRuntimeTool = getToolFromRuntime(currentRuntime)
-    const currentApp = toolLabel => getCurrentApp(getAppType(toolLabel))(apps)
+    const currentApp: any = toolLabel => getCurrentApp(getAppType(toolLabel))(apps)
+
+    //TODO: Bring in as props from Analyses
+    const { loadedState, create, pendingCreate }: AnalysisFileStore = useAnalysisFiles()
```

Comment:
Hmm... is this entirely true though? I think we will still need to bring in the entire `AnalysisFileStore`, even iif iit is from the parent, since a spinner in this file depends on the `loadedState` and the `pendingCreate`. With this in mind, I  think we can modify the above TODO to reflect this, and remove the below TODO

Reply:
This is one thing I dislike about LoadedState. Many data fetching hooks start loading immediately. In those cases, NoneState isn't used and requires jumping through these hoops to access state.